### PR TITLE
Add getLastTotalResultsCount()

### PR DESCRIPTION
**Pull request by martin-rueegg**
Add a function `getLastTotalResultsCount()` to be able to retrieve the `totalResultsCount` that is included in the result.

### CODE REVIEW

**Review by Aternus — CHANGES_REQUESTED** [review]
Please add unit tests related to new functionality.

File: src/Client.php [hunk 1]
```diff
@@ -266,16 +273,34 @@ public function __call(string $endpoint, array $params = [])
...
         $root_property = $this->endpoints[$endpoint];
         if ($root_property !== false && property_exists($response_object, $root_property)
         ) {
+            if ( $this->lastTotalResultsCount === null && is_array($response_object->{$root_property})) {
+                $this->lastTotalResultsCount = count( $response_object->{$root_property} );
+            }
             return $response_object->{$root_property};
         }
 
+        if ( $this->lastTotalResultsCount === null && is_array($response_object)) {
```
Comment (Aternus) [inline src/Client.php]: Looks like there is some duplication that can be generalized...

```php
if ( $this->lastTotalResultsCount === null) {
  if ( $rootProperyExists ) {
    // case A
  } else {
    // case B
  }
}
```

Comment (martin-rueegg) [reply src/Client.php]: Well, I have just extended the existing logic which used a return statement in the if-clause. Hence, the fallback mechanism had to be implemented twice.

I have now simplified the logic by removing the first return statement and instead point the $response_object to it's child $root_property.

File: src/Client.php [hunk 2]
```diff
@@ -266,16 +273,32 @@ public function __call(string $endpoint, array $params = [])
             );
         }
 
+
```
Comment (Aternus) [inline src/Client.php]: Please remove unneeded line-breaks.

File: tests/ClientTest.php [hunk 3]
```diff
@@ -46,6 +46,64 @@ public function testGetSupportedEndpoints()
         $this->assertContains('wikipediaSearch', $endpoints);
     }
 
+    public function testGetLastTotalResultsCount()
+    {
+
+        $g =& $this->client;
+
+        // make sure we start from scratch
+        $this->setUp();
```
Comment (Aternus) [inline tests/ClientTest.php]: Please read on Fixtures: https://phpunit.readthedocs.io/en/7.3/fixtures.html

File: tests/ClientTest.php [hunk 4]
```diff
@@ -46,6 +46,64 @@ public function testGetSupportedEndpoints()
         $this->assertContains('wikipediaSearch', $endpoints);
     }
 
+    public function testGetLastTotalResultsCount()
```
Comment (Aternus) [inline tests/ClientTest.php]: Looks like this test should be split into a couple of smaller ones, each having the appropriate title (per your comments).

File: tests/ClientTest.php [hunk 5]
```diff
@@ -46,6 +46,64 @@ public function testGetSupportedEndpoints()
...
+        ]);
+
+        $this->assertIsArray($arr);
+        $this->assertEmpty($arr);
+
+        $total = $g->getLastTotalResultsCount();
+
+        $this->assertIsInt($total);
+        $this->assertEquals(0, $total);
+
```
Comment (Aternus) [inline tests/ClientTest.php]: Please remove unneeded line breaks. I'll see why PHPCS isn't catching up on this automatically 😵

**Review by Aternus — CHANGES_REQUESTED** [review]
Please see the comments I wrote. Thank you!

File: tests/ClientTest.php (hunk 3 above)
Comment (Aternus) [reply tests/ClientTest.php]: Also, I think it's more readable to use `$this->client` directly. What's your toughts?

**Comment by cedvan** [comment]
:+1:

File: tests/ClientTest.php (hunk 3 above)
Comment (martin-rueegg) [reply tests/ClientTest.php]: > 
> 
> Also, I think it's more readable to use `$this->client` directly. What's your toughts?

I have only copied what was already in the code at [line 135](https://github.com/Aternus/geonames-client/blob/99f11baf551aebde84ea81414cbce77a7848ff04/tests/GeoNames/ClientTest.php#L135) ... ;-)

But if we split the test in multiple test, as suggested by your comment above, then we only reference the client once per test and a local variable doesn't make any sense ...

Comment (martin-rueegg) [reply tests/ClientTest.php]: > 
> 
> Please read on Fixtures: https://phpunit.readthedocs.io/en/7.3/fixtures.html

Ok, didn't realise that. Thanks for the hint. So I can safely remove the setUp()-call! 👍

**Comment by martin-rueegg** [comment]
I have updated the source according to your requests. I have also merged your v2.0.0 into the code to resolve the conflicts.

**Comment by Aternus** [comment]
Looks like this PR is already part of: https://github.com/Aternus/geonames-client/pull/1

If so please delete this one so we can focus on finalizing the other.

**Comment by martin-rueegg** [comment]
Unfortunately, the changes of this PR where accidentally merged into the branch of #1. This has now been fixed and both PRs are clean. Your requests for changes have been applied here.

**Comment by cedvan** [comment]
Hi, I need get total results count, can you merge this PR please ?

**Review by Aternus — APPROVED** [review]

**Pull request merged**